x-pack/apm-server/main.go

```diff
@@ -117,6 +119,23 @@ func newTailSamplingProcessor(args beater.ServerParams) (*sampling.Processor, er
 		return nil, errors.Wrap(err, "failed to create Elasticsearch client for tail-sampling")
 	}
 
+	if entries, err := os.ReadDir(oldTailSamplingStorageDir); err == nil {
+		var newest time.Time
+		for _, de := range entries {
+			if i, err := de.Info(); err == nil {
+				if t := i.ModTime(); t.After(newest) {
+					newest = t
+				}
+			}
+		}
+
+		if time.Since(newest) > tailSamplingConfig.TTL {
```

Comment:
```suggestion
		if newest.IsZero() || time.Since(newest) > tailSamplingConfig.TTL {
```

If there are no entries, we should still delete the directory.